pkg/apis/istio/v1alpha2/values/values_types.proto

```diff
@@ -0,0 +1,911 @@
+// Copyright 2019 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+
+syntax = 'proto3';
+
+package values;
+
+// Package-wide variables from generator "generated".
+option go_package = "v1alpha2";
+
+// AddonIngressConfig is described in istio.io documentation.
+message AddonIngressConfig {
+  bool enabled = 1;
+
+  repeated string hosts = 2;
+}
+
+// ArchConfig is described in istio.io documentation.
+message ArchConfig {
+  // Sets pod scheduling weight for amd64 arch
+  uint32 amd64 = 1;
+
+  // Sets pod scheduling weight for ppc64le arch.
+  uint32 ppc64le = 2;
+
+  // Sets pod scheduling weight for s390x arch.
+  uint32 s390x = 3;
+}
+
+// CNIConfig is described in istio.io documentation.
+message CNIConfig {
+  bool enabled = 1;
+}
+
+// CPUTargetUtilizationConfig is described in istio.io documentation.
+message CPUTargetUtilizationConfig {
+  int32 targetAverageUtilization = 1;
+}
+
+// CertManagerConfig is described in istio.io documentation.
+message CertManagerConfig {
+  bool enabled = 1;
+
+  string hub = 2;
+
+  // 	NodeSelector map[string]interface{} `json:"nodeSelector"`
+  ResourcesConfig resources = 3;
+
+  string tag = 4;
+}
+
+// CoreDNSConfig is described in istio.io documentation.
+message CoreDNSConfig {
+  string coreDNSImage = 1;
+
+  string coreDNSPluginImage = 2;
+
+  bool enabled = 3;
+
+  // 	NodeSelector       map[string]interface{} `json:"nodeSelector,omitempty"`
+  uint32 replicaCount = 4;
+}
+
+// DefaultPodDisruptionBudgetConfig is described in istio.io documentation.
+message DefaultPodDisruptionBudgetConfig {
+  // k8s PodDisruptionBudget settings.
+  bool enabled = 1;
+}
+
+// DefaultResourcesConfig is described in istio.io documentation.
+message DefaultResourcesConfig {
+  // k8s resources settings.
+  ResourcesRequestsConfig requests = 1;
+}
+
+// EgressGatewayConfig is described in istio.io documentation.
+message EgressGatewayConfig {
+  bool autoscaleEnabled = 1;
+
+  uint32 autoscaleMax = 2;
+
+  uint32 autoscaleMin = 3;
+
+  string connectTimeout = 4;
```

Comment:
Could we make this a more specific type so that it validates the field during unmarshal? e.g. have a look at similar fields in k8s to see if we could reuse.

Reply:
yep we can do that. actually it looks like k8s doesn't create specific type for similar fields but I might be missing something

Reply:
ok, i guess it is using dynamic validation on those fields.